lib/backend/model/bgpconfig.go

```diff
@@ -112,7 +112,7 @@ func (key NodeBGPConfigKey) defaultDeletePath() (string, error) {
 }
 
 func (key NodeBGPConfigKey) defaultDeleteParentPaths() ([]string, error) {
-	return nil, nil
+	return []string{fmt.Sprintf("/calico/bgp/v1/host/%s", key.Nodename)}, nil
```

Comment:
This isn't the correct fix.  You really need to have code in the node handling to explicitly delete the BGP peer node sub-tree.  Have a look in node.go in the client directory.  As part of the delete process we also delete the IPAM node sub-tree - I think we want something similar for the bgp sub tree, so perhaps define a BGP Node model that is just used for deletion purposes.